elijah/README.md

```diff
@@ -0,0 +1,111 @@
+
+* match(valA,valB): given card values _valA_ and _valB_ (both of which should be found in _values()_), return true if they match as a pair, or false otherwise.
+
+* display(val): given card value _val_, return a string which represents that card.  If your card values are already strings, this method could merely return _val_, but if your card values are objects, you'll need to generate a string version (e.g. the _name()_ or _shortName()_ of playing-card objects)
+
+For example, if you decide to use regular playing cards in your game, your _values()_ could be a set of 52 card objects generated by your _Card_ constructor.  In that case, an appropriate _match()_ function might return true if two cards are the same rank and same color, creating 26 matching pairs, and _display()_ might return the card's name.
+
+**2) MemoryGame** _[10%]_
+
+The _MemoryGame_ module (a.k.a. the "Model") represents the rules and status of a game (i.e. this positions of the cards and which ones have been matched).  It has an API (command-line interface) but otherwise has no commitment to any graphical representation nor any particular set of cards.
+The game's "board" will be a single row of cards which are identified by a single position number (the GUI is what displays it as a 2D grid).
+
+Each call to the constructor `MemoryGame(cardset)` will construct one game instance using the cards represented in `cardset`, an instance of _MemoryCards_.
+Each _MemoryGame_ instance must keep track of whether and where any card is face up, where any matching cards have been removed, and where unmatched cards remain.
+It will have these methods:
+
+* `reset()` replaces all removed cards, reshuffles the entire board, and rebuilds the board face-down.
+
+* `faceupWhere()` returns the position (a number) of the one face-up card (if any), otherwise returns _false_.
+
+* `faceupValue()` returns the raw value of the one face-up card (if any), otherwise _false_.
+
+* `remaining()` returns an array of the positions of all cards still on the board, including _faceupWhere()_ if any.
+
+* `lift(where)` attempts to lift a card.  If there is a face-down card at position _where_ (a single number), return its display value _(by calling `cardset.display(val)` callback on the card's raw value)_; otherwise return _false_.  If there is not currently a face-up card, leave this card face-up.  If there's already a face-up card, do one of the following:
+
+	* If this card and the face-up card match (according to `cardset.match()`), remove both from the board.  If all pairs are removed from the board, you win the game.
+
+	* If there is no match, leave both cards in place and turn them face down.
+
+It should be possible (though inconvenient) to play an entire game through the console, one `lift()` call at a time.
+
+_This module has been written for you!  You will need to decipher how it works and decide how to integrate your GUI with it._
+
+
+**3) MemoryGUI** _[80%]_
```

Comment:
Nice work with your GUI. Code is clean and easy to read, and you have good comments showing the purpose of particular functions. I like the score count as added functionality. Above and beyond!

Some critiques:

Right now you can click a bunch of different cards in a row, I think I managed to get 5 cards showing at once by quickly clicking through them. You might lower the delay time in hiding the cards, or add in functionality that disables clicking if two cards are showing at once.

For your score counter, keep the score variable in your game's logic, not in your GUI. Remember, the gui doesn't really know or do anything except present what the logic tells it to. Your event listener starting at line 69 is giving your GUI too much power. Otherwise this is a great addition.